castle/cms/_scripts/report-users.py

```diff
@@ -0,0 +1,132 @@
+"""
+The data submitted to a GELF endpoint logger for this is using a schema that
+looks something like this:
+
+class UserAccess(BaseModel):
+    appname: str
+    site: str
+    logid: str
+    reportid: str
+    username: str
+    group: str
+    permissions: list[str]
+
+In this case, Plone (by way of Zope and AccessControl) assigns *permissions*
+to *roles*, and *roles* to *users* in a 3-layer approach.
+
+Therefore, in the above model, the following are mapped to an individual record:
+
+    - username == user.getUserName()
+    - group == a single 
+"""
+from argparse import ArgumentParser
+import csv
+from datetime import datetime
+import logging
+import os
+import sys
+import uuid
+
+from AccessControl.SecurityManagement import newSecurityManager
+from plone import api
+from Products.CMFPlone.interfaces.siteroot import IPloneSiteRoot
+from tendo import singleton
+from zope.component.hooks import setSite
+
+from castle.cms.gelf import GELFHandler
+
+
+
+logger = logging.getLogger("Plone")
+
+
+APP_NAME = "castle.cms"
+LOG_ID = "UserGroupMatrix"
+gelflogger = logging.getLogger(LOG_ID)
+gelfhandler = GELFHandler()
+gelfformatter = logging.Formatter("%(asctime)s %(name)s [%(levelname)s] %(message)s")
+gelfhandler.setFormatter(gelfformatter)
+gelflogger.addHandler(gelfhandler)
+
+
+def get_args():
+    parser = ArgumentParser(
+        description='Get a report of permissions by role and user')
+    parser.add_argument('--site-id', dest='site_id', default='Castle')
+    args, _ = parser.parse_known_args()
+    return args
+
+
+def report_on_users(site):
+    extras = {}
+
+    extras["appname"] = APP_NAME
+    extras["site"] = site.id
+    extras["logid"] = LOG_ID
+    extras["reportid"] = uuid.uuid4()
+
+    users = api.user.get_users()
+    for user in users:
+        extras["username"] = user.getUserName()
```

Comment:
this will just overwrite each previous username, correct? Is that the desired behavior?

Reply:
After reading through the code, it seems like the desire is just to log this data once and then not persist it in any way, so this seems like it should be fine - i'll just leave this comment in case...whatever